pysmiles/smiles_helper.py

```diff
@@ -365,34 +367,39 @@ def valence(atom):
     Returns
     -------
     list[int]
-        The valences for the given atom.
+        The valences for the given atom. Returns an empty list if we don't know.
     """
     element = atom.get("element", '*')
     if element == '*':
-        electrons = 0
-    else:
-        electrons = PTE[element.capitalize()]['AtomicNumber']
-    electrons -= atom.get('charge', 0)
+        return []
 
-    if electrons < 0:
-        raise ValueError(f"Atom {atom} has a negative number of electrons: {electrons}")
-
-    # Let's start by filling complete shells:
-    for shell_idx, shell in enumerate(ORBITAL_SIZES):
-        shell_size = sum(shell)
-        if shell_size <= electrons:
-            electrons -= shell_size
-        else:
-            break
-    else:  # nobreak
-        raise ValueError(f'Too many electrons for sanity for {atom}: {electrons+sum(map(sum, ORBITAL_SIZES))}')
-
-    # Any electrons we have leftover we distribute over the orbitals. First 1
+    electrons = PTE[element.capitalize()]['AtomicNumber']
+    electrons -= atom.get('charge', 0)
+    if not electrons:
+        return [0]
+    try:
+        pte_data = PTE[electrons]
+    except KeyError as error:
+        raise ValueError(f"Can't figure out an electron configuration for {atom}"
+                         f" with {electrons} electrons") from error
+    else:
+        electron_config = pte_data['ElectronConfiguration']
```

Comment:
the else seems not needed here am I wrong?

Reply:
True!